tools/GacInstallTool/Program.cs

```diff
@@ -0,0 +1,39 @@
+// <copyright file="Program.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace GacInstallTool;
+
+internal class Program
+{
+    public static void Main(string[] args)
+    {
+        if (args.Length == 0)
+        {
+            throw new ArgumentNullException(nameof(args));
+        }
+
+        var dir = args[0];
+        var publisher = new System.EnterpriseServices.Internal.Publish();
+        var files = Directory.GetFiles(dir, "*.dll");
+
+        foreach (var file in files)
+        {
+            publisher.GacInstall(file);
```

Comment:
This API can fail silently, it is worth commenting about it.

Reply:
Any way to catch some warnings etc? since it's a .net framework app, it probably can integration into another windows built in service that reports those errors?

Reply:
I think you would have to check for Windows Event logs 😬 
See: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/1984

Reply:
Seems doable > 
https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.eventlog.entrywritten?redirectedfrom=MSDN&view=netframework-4.6.1

Reply:
I think @pjanotti just suggested to give a comment in code. We can improve this later.